### DISCUSSION TITLE
DEBUGV macro usage in LittleFS and other libs, controlled by DEBUG_RP2040_PORT

### DISCUSSION THREAD

**Discussion by Yveaux** [earlephilhower/arduino-pico#2542] [Q&A]
Hi there! In one of my projects I'm using the Littlefs library. Looking at the LittleFS code the DEBUGV macro is, in general, used to log info related to errors. There is one exception though, when closing a file https://github.com/earlephilhower/arduino-pico/blob/fdd755715655bd9ddd6a3d7d6653be5a177c8e6b/libraries/LittleFS/src/LittleFS.h#L468

In other libraries the `DEBUGV` macro is also used for generic debugging, like in
https://github.com/earlephilhower/arduino-pico/blob/fdd755715655bd9ddd6a3d7d6653be5a177c8e6b/cores/rp2040/PIOProgram.cpp#L68

The output from DEBUGV can be suppressed, by undefining the `DEBUG_RP2040_PORT` macro
https://github.com/earlephilhower/arduino-pico/blob/fdd755715655bd9ddd6a3d7d6653be5a177c8e6b/cores/rp2040/debug_internal.h#L24

I am however using the `DEBUG_RP2040_PORT` macro to define where serial debugging ends up (nice), however the logs are spammed with rather useless lines like `lfs_file_close: fd=0x200abcde` when doing some file IO. I do appreciate the DEBUGV logs in case things go wrong so I don't want to suppress all DEBUGV output.

I see a few potential improvements here:
- agree that DEBUGV is used for erroneous situations only and use a different debug macro for generic debug output, which can be disabled individually. Does such macro already exist?
- don't use `DEBUG_RP2040_PORT` to en-/disable the output of library debugging. Is there a better fit macro to control it?

I found a more generic discussion about a logging library in #2066, but that doesn't deal with the strategy of using debugging macro's.

Any thoughts?

**Comment by Yveaux** [comment]
Nobody? Maybe @earlephilhower can add his 5 cents?

**Comment by djpearman** [comment]
Hi,

Just a quick response from me: I, too, have been mildly inconvenienced by the rather useless lines, but was too lazy to do anything about it. So, for the record, I wholeheartedly agree with the removal of the DEBUGV line when closing LittleFS files merged in #2554.

Thank you!
